src/Components/CreateImageWizardV2/utilities/requestMapper.tsx

```
@@ -0,0 +1,164 @@
+  switch (imageType) {
+    case 'aws':
+      if (selectAwsShareMethod(state) === 'sources')
+        return { share_with_sources: [selectAwsSource(state)?.id || ''] };
+      else return { share_with_accounts: [selectAwsAccountId(state)] };
+    case 'gcp': {
+      let googleAccount: string = '';
+      if (selectGcpShareMethod(state) === 'withGoogle') {
+        const gcpEmail = selectGcpEmail(state);
+        switch (selectGcpAccountType(state)) {
+          case 'google':
+            googleAccount = `user:${gcpEmail}`;
+            break;
+          case 'service':
+            googleAccount = `serviceAccount:${gcpEmail}`;
+            break;
+          case 'group':
+            googleAccount = `group:${gcpEmail}`;
+            break;
+          case 'domain':
+            googleAccount = `domain:${gcpEmail}`;
+        }
+        return { share_with_accounts: [googleAccount] };
+      } else {
+        // TODO: GCP withInsights is not implemented yet
+        return {};
+      }
+    }
+  }
+  return {};
+};
+
+const getCustomizations = (state: RootState, orgID: string): Customizations => {
+  return {
+    containers: undefined,
+    directories: undefined,
+    files: undefined,
+    subscription: getSubscription(state, orgID),
+    packages: undefined,
+    payload_repositories: undefined,
+    custom_repositories: selectCustomRepositories(state),
+    openscap: undefined,
+    filesystem: undefined,
+    users: undefined,
+    services: undefined,
+    hostname: undefined,
+    kernel: undefined,
+    groups: undefined,
+    timezone: undefined,
+    locale: undefined,
+    firewall: undefined,
+    installation_device: undefined,
+    fdo: undefined,
+    ignition: undefined,
+    partitioning_mode: undefined,
+    fips: undefined,
+  };
+};
+
+const getSubscription = (state: RootState, orgID: string): Subscription => {
+  const initialSubscription = {
+    'activation-key': selectActivationKey(state) || '',
```

Similar to the discussion around the AWS source ID, I don't think we want to use an empty string here because if we get to this point and there is no activation key... it is a bug. We should have validation on the activation keys step that prevents that from ever happening, so if it does, I think I would prefer that we just fail catastrophically rather than failing 'silently' by building an image that won't have a working subscription.